testsuite/MDAnalysisTests/coordinates/base.py

```diff
@@ -942,61 +926,61 @@ def test_copy(self, func, ts):
+        self._check_independent(self.name, ts)
+        self._check_copy_slice_indices(self.name, ts)
+        self._check_copy_slice_slice(self.name, ts)
+
+    def test_bad_slice(self, some_ts):
         sl = ['this', 'is', 'silly']
         with pytest.raises(TypeError):
-            ts.copy_slice(sl)
+            some_ts.copy_slice(sl)
 
-    def test_bad_copy_slice(self):
-        for p, v, f in itertools.product([True, False], repeat=3):
-            if not any([p, v, f]):
-                continue
-            yield self._check_bad_slice, p, v, f
+    def test_from_timestep(self, some_ts):
+        ts = some_ts
+        ts2 = self.Timestep.from_timestep(ts)
+
+        assert_timestep_almost_equal(ts, ts2)
 
     def _get_pos(self):
         # Get generic reference positions
         return np.arange(30).reshape(10, 3) * 1.234
 
     def _check_ts_equal(self, a, b, err_msg):
-        assert_(a == b, err_msg)
-        assert_(b == a, err_msg)
-
-    def test_check_equal(self):
-        for p, v, f in itertools.product([True, False], repeat=3):
-            if not any([p, v, f]):
-                continue
-
-            ts1 = self.Timestep(self.size,
-                                positions=p,
-                                velocities=v,
-                                forces=f)
-            ts2 = self.Timestep(self.size,
-                                positions=p,
-                                velocities=v,
-                                forces=f)
-            if p:
-                ts1.positions = self.refpos.copy()
-                ts2.positions = self.refpos.copy()
-            if v:
-                ts1.velocities = self.refvel.copy()
-                ts2.velocities = self.refvel.copy()
-            if f:
-                ts1.forces = self.reffor.copy()
-                ts2.forces = self.reffor.copy()
-
-            yield (self._check_ts_equal, ts1, ts2,
-                   'Failed on {0}'.format(self.name))
+        assert a == b, err_msg
+        assert b == a, err_msg
+
+    @pytest.mark.parametrize('p, v, f', filter(any,
+                                               itertools.product([True, False],
+                                                                 repeat=3)))
+    def test_check_equal(self, p, v, f):
+        ts1 = self.Timestep(self.size,
+                            positions=p,
+                            velocities=v,
+                            forces=f)
+        ts2 = self.Timestep(self.size,
+                            positions=p,
+                            velocities=v,
+                            forces=f)
+        if p:
+            ts1.positions = self.refpos.copy()
+            ts2.positions = self.refpos.copy()
+        if v:
+            ts1.velocities = self.refvel.copy()
+            ts2.velocities = self.refvel.copy()
+        if f:
+            ts1.forces = self.reffor.copy()
+            ts2.forces = self.reffor.copy()
+
+        self._check_ts_equal(ts1, ts2, '')
```

Comment:
Should inline `_check_ts_equal`

Reply:
So you mean write directly  

```python
assert ts1 == ts2
assert ts2 == ts1
```

We use the function in two separate places and it does a nice assert that the equal is commutative. Maybe better we make it a stand alone function (not a class method) named `assert_ts_equal`

```python
def assert_ts_equal(ts1, ts2, msg=''):
    assert ts1 == ts2, msg
    assert ts2 == ts1, msg
```

Reply:
I can only see it used once?  Either way yeah, let's split it out into an assert function

Reply:
https://github.com/MDAnalysis/mdanalysis/search?q=_check_ts_equal&type=Code&utf8=%E2%9C%93

The other usage is in a different file. That makes it hard to see. I admit.